libraries/AC_Avoidance/AC_Avoid.cpp

```diff
@@ -1150,6 +1151,11 @@ void AC_Avoid::adjust_velocity_proximity(float kP, float accel_cmss, Vector3f &d
     const Vector2f desired_back_vel_cms_xy = quad_1_back_vel + quad_2_back_vel + quad_3_back_vel + quad_4_back_vel;
     const float desired_back_vel_cms_z = max_back_vel_z + min_back_vel_z;
 
+    if (safe_vel == safe_vel_copy && desired_back_vel_cms_xy.is_zero() && is_zero(desired_back_vel_cms_z)) {
+        // proximity avoidance did nothing, no point in doing the calculations below
```

Comment:
I think we should probably zero the backup velocity.  If not then it's left unchanged from what the caller passed in.  We know that our backup velocity should be zero so let's not make assumptions that the caller will have passed in a zero vector.

Reply:
In another PR perhaps we should look at all the other cases as well, i.e fences, beacons etc. I have done what you've asked for this PR.